pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +115,108 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoints reflect the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
+
+	// State information of the predictor's model.
+	ModelState *ModelStateContainer `json:"modelState,omitempty"`
+
+	// Details of last failure, when load of target model is failed or blocked
+	// +optional
+	LastFailureInfo *FailureInfo `json:"lastFailureInfo,omitempty"`
+
+	// Model replica information of the predictor's model.
+	ModelCopies *ModelCopies `json:"modelCopies,omitempty"`
+}
+
+type ModelStateContainer struct {
```

Comment:
should `TransitionStatus` and `lastFailureInfo` be inside `ModelStateContainer`? seems like they are all related to the model state.

Reply:
Potentially, although the intent of the ModelStateContainer is to specifically keep the state fields (i.e. `activeModelState` and `targetModelState`)  next to each other if a user describes the resource. Adding the other fields would leave us in the same boat. Also, I'm wondering if modelStatus and modelStates are too similar. Would just `states` and `copies` be more succinct?

```yaml
modelStatus:
    transitionStatus: UpToDate
    states:
        activateModelState: Loaded
        targetModelState: 
    copies:
        failedCopies: 0
        totalCopies: 2
```

Reply:
Also rename `ModelStateContainer` to `ModelTransitionState` ?

Reply:
I'm not sure about this one, these reflect states of individual models/revisions rather than the state of the transition (which is what the separate `transitionStatus` field indicates). In the most common case, it would just reflect the state of the the one model backing the predictor (loaded / not loaded / loading / failed).

How about `ModelRevisionStates`?

Reply:
Yea, that makes sense. Updated.